cli/graphql.md

```diff
@@ -899,6 +899,100 @@ The generated resolvers would be protected like so:
+}
+```
+
+**If you deployed your function without amplify**
+
+If you deployed your API without amplify then you must provide the full Lambda function name. If we deployed the same function with the name **echofunction** then you would have:
+
+```
+type Query {
+  echo(msg: String): String @function(name: "echofunction")
+}
+```
+
+**Structure of the AWS Lambda function event**
+
+When writing lambda function's that are connected via the `@function` directive, you can expect the following structure for the AWS Lambda event object.
+
+```javascript
+exports.handler = function (event) {
+  console.log(event);
+  // {
+  //   "typeName": "Query", /* Filled dynamically based on @function usage location */
+  //   "fieldName": "echo", /* Filled dynamically based on @function usage location */
+  //   "arguments": { /* GraphQL field arguments via $ctx.arguments */ },
+  //   "identity": { /* AppSync identity object via $ctx.identity */ },
+  //   "source": { /* The object returned by the parent resolver. E.G. if resolving field 'Post.comments', the source is the Post object. */ },
+  //   "request": { /* AppSync request object. Contains things like headers. */ },
+  //   "prev": { /* If using the built-in pipeline resolver support, this contains the object returned by the previous function. */ },
+  // }
+  context.done(null, event.arguments.msg)
+};
+```
+
+**Calling functions in different regions**
+
+By default, we expect the function to be in the same region as the amplify project. If you need to call a function in a different (or static) region, you can provide the **region** argument.
+
+```
+type Query {
+  echo(msg: String): String @function(name: "echofunction", region: "us-east-1")
+}
+```
+
+Calling functions in different AWS accounts is not supported via the @function directive but is supported by AWS AppSync.
+
+**Chaining functions**
+
+The @function directive supports AWS AppSync pipeline resolvers. That means, you can chain together multiple functions such that they are invoked in series when your field's resolver is invoked. To create a pipeline resolver that calls out to multiple AWS Lambda functions in series, use multiple `@function` directives on the field.
+
+```
+type Mutation {
+  doSomeWork(msg: String): String @function(name: "worker-function") @function(name: "audit-function")
+}
+```
+
+In the example above when you run a mutation that calls the `Mutation.doSomeWork` field, the **worker-function** will be invoked first then the **audit-function** will be invoked with an event that contains the results of the **worker-function** under the **event.prev.result** key. The **audit-function** would need to return **event.prev.result** if you want the result of **worker-function** to be returned for the field.
+
+#### Generates
+
+The `@function` directive generates these resources as necessary:
+
+1. An AWS IAM role that has permission to invoke the function as well as a trust policy with AWS AppSync.
```

Comment:
Note that if the customer wishes to interact with other AWS resources using this function they will need to edit the IAM Execution Policy for that Lambda manually in the AWS Console.

BTW we should think about that in the future - it would be nice if we can let them augment in a friendly way within the CLI so that subsequent `amplify push` events don't blow away the updated policy. cc @kaustavghosh06

Reply:
@undefobj Can you please explain what you mean by

> if the customer wishes to interact with other AWS resources using this function they will need to edit the IAM Execution Policy for that Lambda manually in the AWS Console.

If I have created the function within Amplify, am I not able to edit `{functionName}-cloudformation-template.json` to attach a PolicyDocument to the function there? Or does the `@function` directive somehow override this policy?